storage-fetch/src/p2p/mod.rs

```diff
@@ -0,0 +1,94 @@
+use std::{fmt::Debug, time::Duration};
+
+use anyhow::bail;
+use futures::StreamExt;
+use libp2p::{
+    noise,
+    request_response::{
+        self, cbor::Behaviour as ReqRespBehaviour, Event as ReqRespEvent, Message, ProtocolSupport,
+    },
+    tcp, yamux, Multiaddr, PeerId, StreamProtocol, Swarm, SwarmBuilder,
+};
+use libp2p_swarm::{NetworkBehaviour, SwarmEvent};
+use serde::{de::DeserializeOwned, Serialize};
+use tracing::{info, instrument};
+
+pub mod resolvers;
+
+/// Creates a temporary P2P node. The node then connects to the specified peer
+/// and submits a request. The future resolves when the response is received or
+/// the error is observed.
+#[instrument]
+pub(crate) async fn request_from_peer_sync<Req, Resp>(
+    protocol: &'static str,
+    (peer_id, peer_multiaddr): (PeerId, Multiaddr),
```

Comment:
You're not using the tuple itself for anything
```suggestion
    peer_id: PeerId, peer_multiaddr: Multiaddr,
```